app/src/main/java/org/listenbrainz/android/ui/screens/login/LoginScreen.kt

```diff
@@ -0,0 +1,111 @@
+package org.listenbrainz.android.ui.screens.login
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.Intent
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.airbnb.lottie.compose.LottieAnimation
+import com.airbnb.lottie.compose.LottieCompositionSpec
+import com.airbnb.lottie.compose.LottieConstants
+import com.airbnb.lottie.compose.rememberLottieComposition
+import kotlinx.coroutines.Dispatchers
+import org.listenbrainz.android.R
+import org.listenbrainz.android.repository.AppPreferences
+import org.listenbrainz.android.repository.AppPreferencesImpl
+import org.listenbrainz.android.util.LBSharedPreferences.STATUS_LOGGED_IN
+import org.listenbrainz.android.viewmodel.LoginViewModel
+
+@Composable
+fun ProfileScreen(
+    context : Context = LocalContext.current,
+    viewModel: LoginViewModel = hiltViewModel(),
+    appPreferences: AppPreferences = AppPreferencesImpl(context)
+) {
+    // Initializing app preferences.
+    viewModel.appPreferences = appPreferences
+    
+    val loginStatus = viewModel.getLoginStatusFlow()
+        .collectAsState(initial = viewModel.appPreferences.loginStatus, context = Dispatchers.Default)
+        .value
+    
+    Column(
+        modifier = Modifier.fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        
+        Box {
+            val comp by rememberLottieComposition(spec = LottieCompositionSpec.RawRes(R.raw.login))
+            LottieAnimation(
+                composition = comp,
+                iterations = LottieConstants.IterateForever,
+                modifier = Modifier.requiredHeightIn(max = 400.dp),
+            )
+        }
+        
+        Button(
+            onClick = {
+                when (loginStatus) {
+                    STATUS_LOGGED_IN -> {
+                        viewModel.logoutUser(context)
+                    }
+                    else -> {
+                        val intent = Intent(context, LoginActivity::class.java)
```

Comment:
Why are we launching an activity to handle the login flow? We could manage it right here.

Reply:
Since LoginAcitivity is a webview, we would need to make DashboardActivity a webview which does not count as an app as per android. Also this now being composable, other people can start working on profile screen too

Reply:
Jetpack compose is still behind. We need an explicit activity to be able to open the web which is not what I would have wanted. No other option anyway for now.